nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java

```diff
@@ -216,14 +329,21 @@ public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFacto
                 bindings.put("log", log);
                 bindings.put("REL_SUCCESS", REL_SUCCESS);
                 bindings.put("REL_FAILURE", REL_FAILURE);
+                // only add map relationships if user has opted in
+                if (this.useDynamicRelationships) {
+                    bindings.put("relationships", getRelationshipsAsMap());
```

Comment:
What about adding the dynamic relationships the same way as REL_SUCCESS and REL_FAILURE? We wouldn't need an extra map if we could just `session.transfer(flowFile, REL_TEST)`, and it would be more consistent with the usage of the existing two relationships. You could put the code below into an `else` clause for `if(this.useDynamicRelationships)`, that way you add a REL_XYZ binding in either case, but it's either set to the Relationship `XYZ` or the PropertyDescriptor `XYZ` depending on the setting of "Use Dynamic Relationships"?